File: Sliding Window/MaximumUniqueSubstring.js
function MaxSubstring(str) {
  const m = new Map();
  let i = 0;
  let j = 0;
  let n = str.length;
  let len = 0;
  let max_len = 0;
  let start = -1;

  while (j < n) {
    let ch = str[j];

    if (m.has(ch) && m.get(ch) >= i) {
      i = m.get(ch) + 1;
      len = j - i;
    }

    m.set(ch, j);
    j++;
    len++;

    if (len > max_len) {
      max_len = len;
      start = i;
    }
  }

  return max_len;
}

const str = "pwwkew";

console.log(MaxSubstring(str));

File: Sliding Window/MaximumUniqueSubstring.cpp
#include<bits/stdc++.h>
using namespace std;

string unique_substring(string str) {
    unordered_map<char, int> m;
    int n = str.size();
    int i=0;
    int j=0;
    int len = 0;
    int max_len = 0;
    int start = -1;
    
    while(j < n) {
        char ch = str[j];
        
        if(m.count(ch) != 0 && m[ch] >= i) {
            i = m[ch] + 1;
            len = j-i;
        }
        
        m[ch] = j;
        j++;
        len++;
        
        if(len > max_len) {
            max_len = len;
            start = i;
        }
    }
    
    return str.substr(start, max_len);
}

int main() {
    string str = "abcabed";
    cout<<unique_substring(str)<<endl;
}